executor/common_linux.h

```diff
@@ -766,6 +766,138 @@ static void initialize_devlink_pci(void)
 #endif
 #endif
 
+#if SYZ_EXECUTOR || SYZ_NIC_VF
+#include <ifaddrs.h>
+#include <linux/ethtool.h>
+#include <linux/sockios.h>
+
+struct vf_intf {
+	char pass_thru_intf[IFNAMSIZ];
+	int ppid; // used by Child
+};
+
+static struct vf_intf the_vf_struct, *vfp = &the_vf_struct;
+
+static void find_vf_interface(void)
+{
+	char cmdline[256];
+	struct ifaddrs* addresses = NULL;
+	int pid = getpid();
+	int ret = 0;
+
+	memset(vfp, 0, sizeof(struct vf_intf));
+
+	if (system("lspci | grep ^00:11.0") != 0)
+		return;
+
+	debug("Checking for VF pass-thru interface.\n");
+	if (getifaddrs(&addresses) == -1) {
+		perror("getifaddrs");
+		return;
+	}
+
+	int fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
+
+	if (fd < 0) {
+		perror("socket");
+		return;
+	}
+	struct ifreq ifr;
+	struct ethtool_drvinfo drvinfo;
+	struct ifaddrs* address = addresses;
+
+	while (address) {
+		debug("ifa_name: %s\n", address->ifa_name);
+		memset(&ifr, 0, sizeof(struct ifreq));
+		strcpy(ifr.ifr_name, address->ifa_name);
+		memset(&drvinfo, 0, sizeof(struct ethtool_drvinfo));
+		drvinfo.cmd = ETHTOOL_GDRVINFO;
+		ifr.ifr_data = (caddr_t)&drvinfo;
+		ret = ioctl(fd, SIOCETHTOOL, &ifr);
+
+		if (ret < 0) {
+			perror("ioctl");
+		} else if (strlen(drvinfo.bus_info)) {
+			debug("bus_info: %s, strlen(drvinfo.bus_info)=%zu\n",
+			      drvinfo.bus_info, strlen(drvinfo.bus_info));
+			if (strncmp(drvinfo.bus_info, "0000:00:11.0",
+				    strlen("0000:00:11.0")) == 0) {
+				if (strlen(address->ifa_name) <= IFNAMSIZ) {
+					strncpy(vfp->pass_thru_intf, address->ifa_name, IFNAMSIZ);
+					vfp->ppid = getpid();
+					break;
+				}
+			}
+		}
+		address = address->ifa_next;
+	}
+	freeifaddrs(addresses);
+	if (!vfp->ppid)
+		goto error;
+
+	sprintf(cmdline, "ip netns del staging_ns.%d", pid);
+	if (system(cmdline))
+		perror("system: ip netns del staging_ns");
+
+	sprintf(cmdline, "ip netns add staging_ns.%d", pid);
```

Comment:
Use `procid` variable instead of real pid. These unique staging_ns.%d won't be cleaned up and will pile up infinitely.
procid are between 0 and procs-1 config parameter. So they will be reused and these "ip netns del" will actually work.

Reply:
got rid of the staging ns